### PR TITLE
Upgrade to React 16 and other js stuff

### PR DESCRIPTION
**Pull request by aljordan82**
## What's this PR do?
Upgrade js stuff
## How should this be manually tested?
`rm -rf gulp/node_modules`
`dky`
Then restart devserver.
All React apps should work (Tables, charts, modals, popovers, etc...)
### Front-End

- [ ] Works on IE 11 and Edge
- [x] Works on Mobile
- [x] Unit tests added or ticket exists to add tests

### CODE REVIEW

**Comment by aljordan82** [comment]
Partner Assessment tables are broken in IE11 and Edge

File: gulp/package.json [hunk 1]
```diff
@@ -74,18 +74,19 @@
...
-    "react-bootstrap-table": "^4.0.5",
-    "react-dom": "^15.5.4",
+    "react": "^16.2.0",
+    "react-bootstrap": "^0.31.5",
+    "react-bootstrap-table": "^4.2.0",
+    "react-dom": "^16.2.0",
     "react-grid-layout": "^0.16.0",
-    "react-onclickout": "2.0.7",
+    "react-onclickout": "^2.0.8",
     "react-redux": "5.0.5",
```
Comment (darrint) [inline gulp/package.json]: Do we want to add a `^` here too?

**Comment by aljordan82** [comment]
@darrint package.json is updated with `^` when needed

**Comment by darrint** [comment]
@aljordan82 On IE11 go to PRM Reports, then pick a report and pick preview. Preview does not show. The console shows that it failed to load reporting.js earlier with "invalid character".

**Pull request merged**